py/desisurvey/nextfield.py

```diff
@@ -241,6 +241,11 @@ def get_next_field(dateobs, skylevel, seeing, transparency, previoustiles,
     #- Trim tiles_array to those within 15 degrees of the meridian
     igood = np.where( (last-15 <= tiles_array['RA']) & (tiles_array['RA'] <= last+15) )[0]
     tiles_array = tiles_array[igood]
+    
+    #- Remove previously observed tiles
+    obs = np.in1d( tiles_array['TILEID'], previoustiles )
+    inotobs = np.where(obs == False)
+    tiles_array = tiles_array[inotobs]
```

Comment:
Good use of array logic instead of a loop, but consider using np.in1d(... invert=True) instead of the np.where line:

    notobs = np.in1d(tiles_array['TILEID'], previoustiles, invert=True)
    tiles_array = tiles_array[notobs]

Note that filtering tiles_array[notobs] can accept either a boolean array (like this case) or an array of integer indices; you don't have to convert the boolean array to indices with np.where before using it to filter the array.